shell/platform/fuchsia/dart_runner/dart_test_component_controller_v2.cc

```diff
@@ -0,0 +1,647 @@
+    if (!dart_utils::MappedResource::LoadFromNamespace(
+            namespace_, data_path_ + "/isolate_snapshot_data.bin",
+            isolate_snapshot_data_)) {
+      return false;
+    }
+    if (!dart_utils::MappedResource::LoadFromNamespace(
+            namespace_, data_path_ + "/isolate_snapshot_instructions.bin",
+            isolate_snapshot_instructions_, true /* executable */)) {
+      return false;
+    }
+  }
+  return CreateIsolate(isolate_data, isolate_instructions);
+#endif  // defined(AOT_RUNTIME)
+}
+
+bool DartTestComponentControllerV2::CreateIsolate(
+    const uint8_t* isolate_snapshot_data,
+    const uint8_t* isolate_snapshot_instructions) {
+  // Create the isolate from the snapshot.
+  char* error = nullptr;
+
+  // TODO(dart_runner): Pass if we start using tonic's loader.
+  intptr_t namespace_fd = -1;
+
+  // Freed in IsolateShutdownCallback.
+  auto state = new std::shared_ptr<tonic::DartState>(new tonic::DartState(
+      namespace_fd, [this](Dart_Handle result) { MessageEpilogue(result); }));
+
+  isolate_ = Dart_CreateIsolateGroup(
+      url_.c_str(), label_.c_str(), isolate_snapshot_data,
+      isolate_snapshot_instructions, nullptr /* flags */, state, state, &error);
+  if (!isolate_) {
+    FX_LOGF(ERROR, LOG_TAG, "Dart_CreateIsolateGroup failed: %s", error);
+    return false;
+  }
+
+  state->get()->SetIsolate(isolate_);
+
+  tonic::DartMessageHandler::TaskDispatcher dispatcher =
+      [loop = loop_.get()](auto callback) {
+        async::PostTask(loop->dispatcher(), std::move(callback));
+      };
+  state->get()->message_handler().Initialize(dispatcher);
+
+  state->get()->SetReturnCodeCallback([this](uint32_t return_code) {
+    return_code_ = return_code;
+    auto ret_status = return_code == 0 ? fuchsia::test::Status::PASSED
+                                       : fuchsia::test::Status::FAILED;
+    fuchsia::test::Result result;
+    result.set_status(ret_status);
+    case_listener_->Finished(std::move(result));
+  });
+
+  return true;
+}
+
+// |fuchsia::test::CaseIterator|
+DartTestComponentControllerV2::CaseIterator::CaseIterator(
+    fidl::InterfaceRequest<fuchsia::test::CaseIterator> request,
+    async_dispatcher_t* dispatcher,
+    fit::function<void(CaseIterator*)> done_callback)
+    : binding_(this, std::move(request), dispatcher),
+      done_callback_(std::move(done_callback)) {}
+
+// |fuchsia::test::CaseIterator|
+static const char kTestCaseName[] = "dart_test_v2";
+void DartTestComponentControllerV2::CaseIterator::GetNext(
+    GetNextCallback callback) {
+  if (get_next_call_count == 0) {
+    fuchsia::test::Case test_case;
+    test_case.set_name(std::string(kTestCaseName));
+    test_case.set_enabled(true);
+    std::vector<fuchsia::test::Case> cases;
+    cases.push_back(std::move(test_case));
+    callback(std::move(cases));
+    get_next_call_count++;
+
+  } else {
+    std::vector<fuchsia::test::Case> cases;
+    callback(std::move(cases));
+    // this would be removed
```

Comment:
Does this mean "this should be removed"? Should we have a TODO here?

Reply:
No, this was commenting on what the done_callback is doing. Also residual from copy of legacy_runner code. Removing.